### PR TITLE
{chem,lib}[foss/2026.1,gfbf/2026.1] RDKit v2026.03.4, Boost.Python-NumPy v1.90.0, Eigen 3.4.1, ...

### PR DESCRIPTION
**Pull request by pavelToman**
(created using `eb --new-pr`)
*tests debugged, sanity_checks_commands and sources updated with help of ChatGPT5.5

### CODE REVIEW

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by pavelToman** [comment]
Test report by @pavelToman
**FAILED**
Build succeeded for 1 out of 2 (total: 26 mins 50 secs) (2 easyconfigs in total)
node4008.donphan.os - Linux RHEL 9.6, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz, 1 x NVIDIA NVIDIA A2, 595.71.05, Python 3.9.21
See https://gist.github.com/pavelToman/63bf4c159a282234a9d9fda265197f64 for a full test report.

File: easybuild/easyconfigs/r/RDKit/RDKit-2026.03.3-foss-2026.1.eb [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+sanity_check_paths = {
+    'files': [f'lib/libRDKit{lib}.{SHLIB_EXT}' for lib in _libs],
+    'dirs': [
+        'include/rdkit', 'lib/python%(pyshortver)s/site-packages/rdkit',
+        'lib/python%(pyshortver)s/site-packages/rdkit-%(version)s.dist-info'
+    ],
+}
+
+sanity_check_commands = [
+    "python -c 'import rdkit.rdBase'",
```
Comment (boegel) [inline easybuild/easyconfigs/r/RDKit/RDKit-2026.03.3-foss-2026.1.eb]:
```suggestion
    "python -s -c 'import rdkit.rdBase'",
```

see also:
- https://github.com/easybuilders/easybuild-easyconfigs/pull/25101

**Comment by boegel** [comment]
test step is failing for RDKit:
```
99% tests passed, 2 tests failed out of 263

Total Test time (real) =  71.07 sec

The following tests FAILED:
	162 - distGeomHelpersCatch (Failed)
	176 - moldraw2DTestCatch (Failed)
Errors while running CTest
```

**Comment by Micket** [comment]
testing new tagbot

**Comment by Micket** [comment]
testing to see if i managed to fix the workflow

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 1 out of 2 (total: 41 mins 41 secs) (2 easyconfigs in total)
vera-icelake-build - Linux Rocky Linux 9.8, x86_64, Intel(R) Xeon(R) Silver 4316 CPU @ 2.30GHz, Python 3.9.25
See https://gist.github.com/Micket/35d8c432856599b869135f43bf883c34 for a full test report.

**Comment by Micket** [comment]
Only failing moldraw2DTestCatch for me. Different checksums on basically everything. A few more tests failing when it generated an empty svg file. I don't know what the original files are, so it's hard to identify what the svgs should contain. It's a checksum, so it might be an extra space, or it might all be wrong.

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by pavelToman** [comment]
Test report by @pavelToman
**SUCCESS**
Build succeeded for 2 out of 2 (total: 2 hours 28 mins 2 secs) (2 easyconfigs in total)
node4007.donphan.os - Linux RHEL 9.6, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz, 1 x NVIDIA NVIDIA A2, 595.71.05, Python 3.9.21
See https://gist.github.com/pavelToman/76ff1de4284719fca0e86cd8878d31c0 for a full test report.

**Comment by pavelToman** [comment]
Test report by @pavelToman
**FAILED**
Build succeeded for 2 out of 3 (total: 29 mins 51 secs) (2 easyconfigs in total)
node4204.shinx.os - Linux RHEL 9.6, x86_64, AMD EPYC 9654 96-Core Processor, Python 3.9.21
See https://gist.github.com/pavelToman/a9d5fc2a4d1b29844c4baa9652820ee3 for a full test report.

```
The following tests FAILED:
        194 - testCIPLabeler (Failed)
```

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by pavelToman** [comment]
Test report by @pavelToman
**SUCCESS**
Build succeeded for 2 out of 2 (total: 28 mins 54 secs) (2 easyconfigs in total)
node4204.shinx.os - Linux RHEL 9.6, x86_64, AMD EPYC 9654 96-Core Processor, Python 3.9.21
See https://gist.github.com/pavelToman/f0b8df2b587c8057bad7e7c2435cbd66 for a full test report.

**Comment by Thyre** [comment]
Test report by @Thyre
**FAILED**
Build succeeded for 5 out of 6 (total: 12 mins 26 secs) (2 easyconfigs in total)
Linux - Linux Arch Linux UNKNOWN, x86_64, AMD Ryzen 7 7800X3D 8-Core Processor (zen4), 1 x AMD Navi 48 [Radeon RX 9070/9070 XT/9070 GRE] (device id: 0x7550, gfx: gfx1201, driver: Linuxversion7.1.2-arch3-1(linux@archlinux)(gcc(GCC)16.1.120260625,GNUld(GNUBinutils)2.46.1)#1SMPPREEMPT_DYNAMICFri,03Jul202623:25:36+0000), 1 x AMD Raphael (device id: 0x164e, gfx: gfx1036, driver: Linuxversion7.1.2-arch3-1(linux@archlinux)(gcc(GCC)16.1.120260625,GNUld(GNUBinutils)2.46.1)#1SMPPREEMPT_DYNAMICFri,03Jul202623:25:36+0000), Python 3.14.6
See https://gist.github.com/Thyre/5706805f5b82fb8cef11bb0160b2a124 for a full test report.

----

```
== sanity checking...
[...]
  >> file 'lib/libRDKitcoordgen.so' found: FAILED
[...]
  >> file 'lib/libRDKitmaeparser.so' found: FAILED
```

File: easybuild/easyconfigs/r/RDKit/RDKit-2026.03.3-foss-2026.1.eb [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+toolchain = {'name': 'foss', 'version': '2026.1'}
+
+source_urls = ['https://github.com/rdkit/rdkit/archive/']
+sources = [f'Release_{_version}.tar.gz']
+checksums = ['21e22e5e6b3a313527256fbde41c757f22d834b19caf2908e3c2dd11061e1fea']
+
+builddependencies = [
+    ('CMake', '4.2.1'),
+    ('Eigen', '5.0.0'),
```
Comment (Thyre) [inline easybuild/easyconfigs/r/RDKit/RDKit-2026.03.3-foss-2026.1.eb]: This isn't picked up correctly:

```
-- Eigen3 version .. found in /data/EasyBuild-develop/software/Eigen/5.0.0-GCCcore-15.2.0/include, but at least version 2.91.0 is required
-- Could NOT find Eigen3 (missing: EIGEN3_VERSION_OK) (Required is at least version "2.91.0")
```

**Comment by Thyre** [comment]
**RDKit:**

Inchi is picked up from the system:
```
-- Found Inchi: /usr/lib/libinchi.so
```

Files are downloaded during `cmake` call:

```
== Using strict rotor definition
Downloading https://github.com/mittinatten/freesasa/releases/download/2.0.3/freesasa-2.0.3.tar.gz...
Copying freesasa2.c to /data/EasyBuild-develop/build/RDKit/2026.03.3/foss-2026.1/rdkit-Release_2026_03_3/External/FreeSASA/freesasa-2.0.3/src/freesasa2.c
Downloading https://github.com/rareylab/RingDecomposerLib/archive/v1.1.3_rdkit.tar.gz...
Downloading https://github.com/ncbi/pubchem-align3d/archive/daefab3.tar.gz...
[...]
== expat turn off warnings as errors when building
Downloading https://github.com/google/fonts/raw/main/ofl/comicneue/ComicNeue-Regular.ttf...
Downloading https://github.com/google/fonts/raw/main/ofl/comicneue/OFL.txt...
```

And Eigen during the build (see above):

```
cd /data/EasyBuild-develop/build/RDKit/2026.03.3/foss-2026.1/rdkit-Release_2026_03_3/External/Eigen && /data/EasyBuild-develop/software/CMake/4.2.1-GCCcore-15.2.0/bin/cmake -DCMAKE_MESSAGE_LOG_LEVEL=VERBOSE -P /data/EasyBuild-develop/build/RDKit/2026.03.3
/foss-2026.1/easybuild_obj/External/Eigen/Eigen-prefix/tmp/Eigen-gitclone.cmake
Cloning into 'eigen'...
/usr/lib/git-core/git-remote-https: /data/EasyBuild-develop/software/cURL/8.17.0-GCCcore-15.2.0/lib/libcurl.so.4: no version information available (required by /usr/lib/git-core/git-remote-https)
```

**Comment by Thyre** [comment]
> Test report by @Thyre
> 
> **FAILED**
> 
> Build succeeded for 5 out of 6 (total: 12 mins 26 secs) (2 easyconfigs in total)
> 
> Linux - Linux Arch Linux UNKNOWN, x86_64, AMD Ryzen 7 7800X3D 8-Core Processor (zen4), 1 x AMD Navi 48 [Radeon RX 9070/9070 XT/9070 GRE] (device id: 0x7550, gfx: gfx1201, driver: Linuxversion7.1.2-arch3-1(linux@archlinux)(gcc(GCC)16.1.120260625,GNUld(GNUBinutils)2.46.1)#1SMPPREEMPT_DYNAMICFri,03Jul202623:25:36+0000), 1 x AMD Raphael (device id: 0x164e, gfx: gfx1036, driver: Linuxversion7.1.2-arch3-1(linux@archlinux)(gcc(GCC)16.1.120260625,GNUld(GNUBinutils)2.46.1)#1SMPPREEMPT_DYNAMICFri,03Jul202623:25:36+0000), Python 3.14.6
> 
> See https://gist.github.com/Thyre/5706805f5b82fb8cef11bb0160b2a124 for a full test report.
> 
> 
> 
> ----
> 
> 
> 
> ```
> 
> == sanity checking...
> 
> [...]
> 
>   >> file 'lib/libRDKitcoordgen.so' found: FAILED
> 
> [...]
> 
>   >> file 'lib/libRDKitmaeparser.so' found: FAILED
> 
> ```

If one checks the logs, the failing libraries are also picked up from the system:

```
libcoordgen.so.3 => /usr/lib/libcoordgen.so.3 (0x00007fdab3865000)
libmaeparser.so.1 => /usr/lib/libmaeparser.so.1 (0x00007fdab34cf000)
```

Also, some Boost libraries are picked up, which do not come from EasyBuild, e.g.

```
libboost_regex.so.1.91.0 => /usr/lib/libboost_regex.so.1.91.0 (0x00007fdab2891000)
libboost_random.so.1.91.0 => /usr/lib/libboost_random.so.1.91.0 (0x00007fdab2c8e000)
libboost_iostreams.so.1.91.0 => /usr/lib/libboost_iostreams.so.1.91.0 (0x00007fdab2ce0000)
```

**Comment by pavelToman** [comment]
Thank you for your reports @Thyre, I will check this as soon as our clusters will be back online.

**Comment by pavelToman** [comment]
@Thyre the commit https://github.com/easybuilders/easybuild-easyconfigs/pull/26331/commits/515641cd74acb3af5fb2077e7b38e4c16776e837 should fix all the problems you reported:
- handling Boost dependency to not pick the system one
- use Eigen3
- linking to libinchi.so, libcoordgen.so.3, libmaeparser.so.1
- downloading sources during the installation

Could you please test it again?

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

File: easybuild/easyconfigs/r/RDKit/RDKit-2026.03.3-foss-2026.1.eb [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    {'RingDecomposerLib-1.1.3_rdkit.tar.gz': '944b5816712a48bbf88aa25d4300ce11871ddf6e971218eac08f90ed2192f715'},
+    {'pubchem-align3d-daefab3.tar.gz': '4871a08104c660890015f27d32a0147fc32df69a8962ca5cd1fe54ef6319c779'},
+    {'chemdraw-1.0.12.tar.gz': 'c2fed7000b592f279773c5a35b0d08dfe93709fa352ab28c90b520dffb974c2b'},
+]
+
+_boost_version = '1.90.0'
+
+builddependencies = [
+    ('CMake', '4.2.1'),
+    ('Eigen', '3.4.0'),
```
Comment (ocaisa) [inline easybuild/easyconfigs/r/RDKit/RDKit-2026.03.3-foss-2026.1.eb]: How is Eigen only a build dependency?...and why the massive drop in version `5.0.0 -> 3.4.0`

Comment (ocaisa) [reply easybuild/easyconfigs/r/RDKit/RDKit-2026.03.3-foss-2026.1.eb]: Ah, it it's a C++ template library

Comment (pavelToman) [reply easybuild/easyconfigs/r/RDKit/RDKit-2026.03.3-foss-2026.1.eb]: Seems RDKit does not works with new Eigen v5 yet: https://github.com/easybuilders/easybuild-easyconfigs/pull/26331#discussion_r3527521789

and it is always the builddep: https://github.com/easybuilders/easybuild-easyconfigs/blob/798833aef8563a267fb10acf5e5d29d75fbe0b08/easybuild/easyconfigs/r/RDKit/RDKit-2025.03.5-foss-2025a.eb#L18

Comment (ocaisa) [reply easybuild/easyconfigs/r/RDKit/RDKit-2026.03.3-foss-2026.1.eb]: At the least, I think this should be 3.4.1: https://gitlab.com/libeigen/eigen/-/releases/3.4.1

Comment (ocaisa) [reply easybuild/easyconfigs/r/RDKit/RDKit-2026.03.3-foss-2026.1.eb]: And it is Eigen3 due to https://github.com/rdkit/rdkit/blob/776403299789871b2aaf9feefb7125ca2037135a/Code/cmake/Modules/FindEigen3.cmake

Comment (pavelToman) [reply easybuild/easyconfigs/r/RDKit/RDKit-2026.03.3-foss-2026.1.eb]: Since the RDKit downloads the v3.4.0 during the installation (when it can not find suitable Eigen) I think the v3.4.0 is right choice here.

File: easybuild/easyconfigs/b/Boost.Python-NumPy/Boost.Python-NumPy-1.90.0-gfbf-2026.1.eb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+easyblock = 'EB_Boost'
+
+name = 'Boost.Python-NumPy'
+version = '1.90.0'
```
Comment (ocaisa) [inline easybuild/easyconfigs/b/Boost.Python-NumPy/Boost.Python-NumPy-1.90.0-gfbf-2026.1.eb:4]: There is release 1.91.0: https://archives.boost.io/release/1.91.0/

File: easybuild/easyconfigs/e/Eigen/Eigen-3.4.0-GCCcore-15.2.0.eb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+name = 'Eigen'
+version = '3.4.0'
```
Comment (ocaisa) [inline easybuild/easyconfigs/e/Eigen/Eigen-3.4.0-GCCcore-15.2.0.eb]: Latest v3 release is 3.4.1

File: easybuild/easyconfigs/r/RDKit/RDKit-2026.03.3-foss-2026.1.eb [hunk 6]
```diff
@@ -0,0 +1,191 @@
+easyblock = 'CMakeMake'
+
+name = 'RDKit'
+version = '2026.03.3'
```
Comment (ocaisa) [inline easybuild/easyconfigs/r/RDKit/RDKit-2026.03.3-foss-2026.1.eb]: 2026.03.4 was released last week, can we attempt a trivial update (if it leads to additional problems I'm ok with sticking with this version)

**Review by ocaisa — CHANGES_REQUESTED** [review]
Some more version comments

**Comment by pavelToman** [comment]
Test report by @pavelToman
**SUCCESS**
Build succeeded for 5 out of 5 (total: 19 mins 0 secs) (5 easyconfigs in total)
node4428.skiddo.os - Linux RHEL 9.6, x86_64, AMD EPYC 9755 128-Core Processor, Python 3.9.21
See https://gist.github.com/pavelToman/bf94d0554de66eb2bfc87547ed7d14a5 for a full test report.

File: easybuild/easyconfigs/r/RDKit/RDKit-2026.03.3-foss-2026.1.eb (hunk 6 above)
Comment (pavelToman) [reply easybuild/easyconfigs/r/RDKit/RDKit-2026.03.3-foss-2026.1.eb]: ok, I am gonna test the new version

File: easybuild/easyconfigs/r/RDKit/RDKit-2026.03.3-foss-2026.1.eb (hunk 3 above)
Comment (pavelToman) [reply easybuild/easyconfigs/r/RDKit/RDKit-2026.03.3-foss-2026.1.eb]: But I can try 3.4.1

File: easybuild/easyconfigs/b/Boost.Python-NumPy/Boost.Python-NumPy-1.90.0-gfbf-2026.1.eb (hunk 4 above)
Comment (pavelToman) [reply easybuild/easyconfigs/b/Boost.Python-NumPy/Boost.Python-NumPy-1.90.0-gfbf-2026.1.eb:4]: Since Boost version is already established for GCCcore/15.2.0, I think there should be only one version per toolchain?

**Comment by pavelToman** [comment]
> Test report by @boegelbot **FAILED** Build succeeded for 4 out of 5 (total: 24 mins 11 secs) (5 easyconfigs in total) jsczen3c1.int.jsc-zen3.fz-juelich.de - Linux Rocky Linux 9.8, x86_64, AMD EPYC-Milan Processor (zen3), Python 3.9.25 See https://gist.github.com/boegelbot/68683e1e8772e459b14dbcf4d1434ca7 for a full test report.

The same issue as I had - seems the old installdir needs to be removed before the new installation then the linking should be ok.
But I can not `rm -r %(installdir)s` before tests - this can make a problem when RDKit is already installed - will not be available for the users.

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3 EB_ARGS="--installpath /tmp/$USER/pr26331"

File: easybuild/easyconfigs/b/Boost.Python-NumPy/Boost.Python-NumPy-1.90.0-gfbf-2026.1.eb (hunk 4 above)
Comment (ocaisa) [reply easybuild/easyconfigs/b/Boost.Python-NumPy/Boost.Python-NumPy-1.90.0-gfbf-2026.1.eb:4]: Ah, sorry, my bad

**Comment by pavelToman** [comment]
Test report by @pavelToman
**SUCCESS**
Build succeeded for 5 out of 5 (total: 30 mins 37 secs) (5 easyconfigs in total)
node4209.shinx.os - Linux RHEL 9.6, x86_64, AMD EPYC 9654 96-Core Processor, Python 3.9.21
See https://gist.github.com/pavelToman/3e0be9de36b4dd0fcd9c1ac7a226c7bc for a full test report.

File: easybuild/easyconfigs/r/RDKit/RDKit-2026.03.3-foss-2026.1.eb (hunk 6 above)
Comment (pavelToman) [reply easybuild/easyconfigs/r/RDKit/RDKit-2026.03.3-foss-2026.1.eb]: version updated

File: easybuild/easyconfigs/e/Eigen/Eigen-3.4.0-GCCcore-15.2.0.eb (hunk 5 above)
Comment (pavelToman) [reply easybuild/easyconfigs/e/Eigen/Eigen-3.4.0-GCCcore-15.2.0.eb]: version updated

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3 EB_ARGS="--installpath /tmp/$USER/pr26331"

**Comment by pavelToman** [comment]
Test report by @pavelToman
**SUCCESS**
Build succeeded for 5 out of 5 (total: 23 mins 47 secs) (5 easyconfigs in total)
node4428.skiddo.os - Linux RHEL 9.6, x86_64, AMD EPYC 9755 128-Core Processor, Python 3.9.21
See https://gist.github.com/pavelToman/bd3646dc022c0772422540e93212a0ff for a full test report.

**Comment by pavelToman** [comment]
Test report by @pavelToman
**FAILED**
Build succeeded for 4 out of 5 (total: 36 mins 51 secs) (5 easyconfigs in total)
node4002.donphan.os - Linux RHEL 9.6, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz, 1 x NVIDIA NVIDIA A2, 595.71.05, Python 3.9.21
See https://gist.github.com/pavelToman/1203ee51444ffb86b7af807e80008d9f for a full test report.

**Comment by pavelToman** [comment]
Test report by @pavelToman
**SUCCESS**
Build succeeded for 5 out of 5 (total: 33 mins 34 secs) (5 easyconfigs in total)
node4209.shinx.os - Linux RHEL 9.6, x86_64, AMD EPYC 9654 96-Core Processor, Python 3.9.21
See https://gist.github.com/pavelToman/3591f0856be6bf2b41d52c4001d0f4d9 for a full test report.

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3 EB_ARGS="--installpath /tmp/$USER/pr26331"

**Review by ocaisa — APPROVED** [review]
LGTM

**Comment by pavelToman** [comment]
Test report by @pavelToman
**SUCCESS**
Build succeeded for 5 out of 5 (total: 33 mins 32 secs) (5 easyconfigs in total)
node4002.donphan.os - Linux RHEL 9.6, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz, 1 x NVIDIA NVIDIA A2, 595.71.05, Python 3.9.21
See https://gist.github.com/pavelToman/32505da24abf99c9d7828e10f20f2025 for a full test report.

**Comment by ocaisa** [comment]
Going in, thanks for refreshing this @pavelToman

**Comment by pavelToman** [comment]
Test report by @pavelToman
**SUCCESS**
Build succeeded for 5 out of 5 (total: 22 mins 43 secs) (5 easyconfigs in total)
node4401.skiddo.os - Linux RHEL 9.6, x86_64, AMD EPYC 9755 128-Core Processor, Python 3.9.21
See https://gist.github.com/pavelToman/8445a10dacfcebb604d67a5fbf308e3e for a full test report.

File: easybuild/easyconfigs/r/RDKit/RDKit-2026.03.4-foss-2026.1.eb [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+pretestopts = 'export LD_LIBRARY_PATH=%(builddir)s/easybuild_obj/lib:${LD_LIBRARY_PATH} && '
+pretestopts += 'export RDBASE=$PWD && '
+pretestopts += 'export PYTHONPATH=$PWD:$PYTHONPATH &&'
+
+runtest = True
+
+testopts = (
+    "-j %(parallel)s -E "
+    "'pythonSourceTests|"  # skip test with known issues
+    "testDistGeomHelpers|"  # platform-sensitive ETKDG/distance-geometry regression checks
```
Comment (boegel) [inline easybuild/easyconfigs/r/RDKit/RDKit-2026.03.4-foss-2026.1.eb:135]: @pavelToman It seems like we don't need to skip this after all, based on the feedback in:
- https://github.com/rdkit/rdkit/issues/9406
We can jump bump the tolerance?

Comment (pavelToman) [reply easybuild/easyconfigs/r/RDKit/RDKit-2026.03.4-foss-2026.1.eb:135]: Fixed in: https://github.com/easybuilders/easybuild-easyconfigs/pull/26580
